tests/systemtests/authz_test.go

```diff
@@ -0,0 +1,840 @@
+//go:build system_test
+
+package systemtests
+
+import (
+	"fmt"
+	"testing"
+	"time"
+
+	"github.com/stretchr/testify/assert"
+	"github.com/stretchr/testify/require"
+	"github.com/tidwall/gjson"
+)
+
+const (
+	msgSendTypeURL       = `/cosmos.bank.v1beta1.MsgSend`
+	msgDelegateTypeURL   = `/cosmos.staking.v1beta1.MsgDelegate`
+	msgVoteTypeURL       = `/cosmos.gov.v1.MsgVote`
+	msgUndelegateTypeURL = `/cosmos.staking.v1beta1.MsgUndelegate`
+	msgRedelegateTypeURL = `/cosmos.staking.v1beta1.MsgBeginRedelegate`
+)
+
+func TestAuthzGrantTxCmd(t *testing.T) {
+	// scenario: test authz grant command
+	// given a running chain
+
+	sut.ResetChain(t)
+	cli := NewCLIWrapper(t, sut, verbose)
+
+	// get validator address which will be used as granter
+	granterAddr := gjson.Get(cli.Keys("keys", "list"), "1.address").String()
+	require.NotEmpty(t, granterAddr)
+
+	// add grantee keys which will be used for each valid transaction
+	grantee1Addr := cli.AddKey("grantee1")
+	grantee2Addr := cli.AddKey("grantee2")
+	require.NotEqual(t, granterAddr, grantee2Addr)
+	grantee3Addr := cli.AddKey("grantee3")
+	require.NotEqual(t, granterAddr, grantee3Addr)
+	grantee4Addr := cli.AddKey("grantee4")
+	require.NotEqual(t, granterAddr, grantee4Addr)
+	grantee5Addr := cli.AddKey("grantee5")
+	require.NotEqual(t, granterAddr, grantee5Addr)
+	grantee6Addr := cli.AddKey("grantee6")
+	require.NotEqual(t, granterAddr, grantee6Addr)
+
+	sut.StartChain(t)
+
+	// query validator operator address
+	rsp := cli.CustomQuery("q", "staking", "validators")
+	valOperAddr := gjson.Get(rsp, "validators.#.operator_address").Array()[0].String()
+
+	grantCmdArgs := []string{"tx", "authz", "grant", "--from", granterAddr}
+	expirationTime := time.Now().Add(time.Minute * time.Duration(10)).Unix()
```

Comment:
I am not able to find any method to increase time of node. So, I just kept it for 10 seconds so that we can make tests wait for that time and check expiration then.